File: all/lcof10-I.fei-bo-na-qi-shu-lie-lcof/Solution.java

class Solution {
    int[] array = new int[101];

    public int fib(int n) {
        if (array[n] != 0)
            return array[n];

        long res = 0;
        if (n == 0) {
            res = 0;
        } else if (n == 1) {
            res = 1;
        } else {
            res = fib(n - 1) + fib(n - 2);
        }
        res %= (Math.pow(10, 9) + 7);
        array[n] = (int) res;

        return (int) res;
    }
}

File: all/lcof10-I.fei-bo-na-qi-shu-lie-lcof/Solution.cpp
#include <math.h>

class Solution
{
public:
    int fib(int n)
    {
        if (n < 2)
            return n;
        int a = 0, b = 1;

        for (int i = 2; i <= n; i++)
        {
            b = a + b;
            a = b - a;
            b = fmod(b, 1000000007);
        }
        return b;
    }
};
